### PR TITLE
Add support for sort and handle streaming of empty datasets on XML Responses

### LINKED ISSUE #2000: Data Endpoint: Support pagination of XML Response

### PR DESCRIPTION
**Pull request by DavisRayM**
### Changes / Features implemented

- Handle streaming empty datasets using XML Renderer
- Add `_date_modified` to the NONE_JSON_FIELD list
- Sort none & filter none JSON fields using the model field

### Steps taken to verify this change does what is intended

- Added tests

### Side effects of implementing this change

- N/A

Closes #2000

### CODE REVIEW

File: onadata/libs/models/sorting.py [hunk 1]
```diff
@@ -21,11 +22,16 @@ def sort_from_mongo_sort_str(sort_str):
...
     _list = []
 
     for field in sort_list:
-        _str = u" json->>%s"
+        field_key = field.lstrip('-')
+        _str = u" json->>%s"\
+            if field_key not in none_json_fields.keys() else\
+            f'"{model_name}"."{none_json_fields.get(field_key)}"'
+
         if field.startswith('-'):
```
Comment (ukanga) [inline onadata/libs/models/sorting.py:35]: Should we add or update the unit tests for the `json_order_by()` function?

File: onadata/libs/models/sorting.py [hunk 2]
```diff
@@ -38,10 +44,12 @@ def json_order_by(sort_list):
...
 
-def json_order_by_params(sort_list):
+def json_order_by_params(sort_list, none_json_fields: Dict = {}):
     params = []
 
     for field in sort_list:
-        params.append(field.lstrip('-'))
+        field = field.lstrip('-')
+        if field not in none_json_fields.keys():
+            params.append(field.lstrip('-'))
```
Comment (ukanga) [inline onadata/libs/models/sorting.py:53]: Should we add or update the unit tests for the `json_order_by_params()` function?

**Review by ukanga — COMMENTED** [review]
Could we also revisit `<submission-item created=` to use `<submission-item dateCreated=` to be more consistent in how the attributes are defined?

File: onadata/libs/models/sorting.py (hunk 2 above)
Comment (DavisRayM) [reply onadata/libs/models/sorting.py:53]: Added in the latest commits

File: onadata/libs/models/sorting.py (hunk 1 above)
Comment (DavisRayM) [reply onadata/libs/models/sorting.py:35]: Added in the latest commits

**Review by ukanga — APPROVED** [review]

**Comment by gstuder-ona** [comment]
@DavisRayM sorry I didn't get to this earlier, but we're using the endpoint from stage successfully now

**Pull request merged**